test/server.js

```diff
@@ -547,8 +549,8 @@ const createTestSuite = ({
                     return requestPromised(opts)
                         .then((response) => {
                             if (useMainProxy) {
-                                expect(response.statusCode).to.eql(500);
-                                expect(response.body).to.match(/with an invalid HTTP status code/);
+                                expect(response.statusCode).to.eql(502);
+                                expect(response.body).to.eql('Bad status!');
```

Comment:
Now instead of rejecting the entire request on invalid status code, the body will be reused and the status code will be set to 502.